apps/server/src/modules/tool/external-tool/uc/external-tool.uc.ts

```diff
@@ -88,8 +95,40 @@ export class ExternalToolUc {
 		return metadata;
 	}
 
-	private async ensurePermission(userId: EntityId, permission: Permission) {
+	private async ensurePermission(userId: EntityId, permission: Permission): Promise<void> {
 		const user: User = await this.authorizationService.getUserWithPermissions(userId);
 		this.authorizationService.checkAllPermissions(user, [permission]);
 	}
+
+	public async getDatasheet(userId: EntityId, externalToolId: EntityId): Promise<Buffer> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkOneOfPermissions(user, [Permission.TOOL_ADMIN, Permission.SCHOOL_TOOL_ADMIN]);
+
+		const externalTool: ExternalTool = await this.externalToolService.findById(externalToolId);
+		const dataSheetData: ExternalToolDatasheetTemplateData =
+			ExternalToolDatasheetMapper.mapToExternalToolDatasheetTemplateData(externalTool, user.firstName, user.lastName);
+
+		const buffer: Buffer = await this.datasheetPdfService.generatePdf(dataSheetData);
+
+		return buffer;
+	}
+
+	public async createDatasheetFilename(externalToolId: EntityId): Promise<string> {
+		const externalTool: ExternalTool = await this.externalToolService.findById(externalToolId);
+		const filename: string = this.buildFilename(externalTool.name);
+
+		return filename;
+	}
+
+	private buildFilename(toolName: string): string {
```

Comment:
this seems unnecessary to make a private function for when the calling function is only 3 lines.